### PR TITLE
Add is_configured to NotificationPlugin

### LINKED ISSUE #2319: Check Plugin.is_configured before trying to rule_notify
This causes errors because a plugin is installed, but not ready to work correctly.

### PR DESCRIPTION
**Pull request by mattrobenolt**
All of the upstream projects that we load in that subclass
NotificationPlugin, already implement is_configured(self, project). So
this adds that API to the base NotificationPlugin class, and enforces
that it's called before trying to send notifications.

Fixes GH-2319

### CODE REVIEW

**Comment by mattrobenolt** [comment]
I've verified all of our plugins implement this correctly.

One plugin, `sentry-twilio` (which I wrote) implemented `is_configured` incorrectly: https://github.com/mattrobenolt/sentry-twilio/commit/4d436a7626a9674ec2f1f6b137f4e2711b00f292

Other than that, every one we install does this correctly.

File: src/sentry/plugins/bases/notify.py [hunk 1]
```diff
@@ -129,7 +129,13 @@ def __is_rate_limited(self, group, event):
             limit=10,
         )
 
+    def is_configured(self, project):
```
Comment (dcramer) [inline src/sentry/plugins/bases/notify.py]: do all notification plugins implement this?

Comment (mattrobenolt) [inline src/sentry/plugins/bases/notify.py]: See my comment. :)

I checked every one that we run on getsentry for it, and they all implement this signature, surprisingly. So my guess is that we used to support this historically, then somehow stopped?

Comment (dcramer) [inline src/sentry/plugins/bases/notify.py]: We never supported it, we just copypasta

Comment (mattrobenolt) [inline src/sentry/plugins/bases/notify.py]: :+1: Then luckily, yeah, every notification plugin we run has this implemented as expected.

**Pull request merged**